packages/angular/src/generators/scam/lib/create-module.ts

```diff
@@ -83,19 +83,55 @@ export function createScam(tree: Tree, schema: Schema) {
     )}`;
 
     tree.write(componentFilePath, updatedComponentSource);
+    exportScam(tree, schema, componentFilePath);
     return;
   }
 
+  const scamFilePath = joinPathFragments(
+    componentDirectory,
+    `${componentNames.fileName}.module.ts`
+  );
+
   tree.write(
-    joinPathFragments(
-      componentDirectory,
-      `${componentNames.fileName}.module.ts`
-    ),
+    scamFilePath,
     createSeparateAngularComponentModuleFile(
       `${componentNames.className}${typeNames.className}`,
       componentFileName
     )
   );
+
+  exportScam(tree, schema, scamFilePath);
+}
+
+function exportScam(tree: Tree, schema: Schema, scamFilePath: string) {
+  if (!schema.export) {
+    return;
+  }
+
```

Comment:
We should check if `projectType === application` and maybe log a warning that the `export=true` was ignored because it only applies to libraries. Otherwise, it will continue and log a warning that doesn't apply to that project.

A test case covering this would be helpful.

Reply:
good idea!